scopes/webpack/modules/fragments/style-preset/style-preset.ts

```diff
@@ -0,0 +1,202 @@
+  /** override the options for resolveUrl loader */
+  resolveUrlOptions?: any;
+};
+
+export type StyleLoadersOptions = CssOptions & CssDialectOptions;
+
+/** loaders for css, including dialects (sass and less), postcss, and css-modules */
+export const makeStyleLoaders = ({
+  styleInjector = 'mini-css-extract-plugin',
+  modules = true,
+
+  cssLoader = require.resolve('css-loader'),
+  postcssLoader = require.resolve('postcss-loader'),
+  resolveUrlLoader = require.resolve('resolve-url-loader'),
+  sassLoader = require.resolve('sass-loader'),
+  lessLoader = require.resolve('less-loader'),
+
+  miniCssOptions,
+  cssLoaderOptions,
+  injectorOptions,
+  postcssOptions,
+  sassLoaderOptions,
+  lessLoaderOptions,
+  resolveUrlOptions,
+}: StyleLoadersOptions) => {
+  const styleLoaders: RuleSetRule[] = [
+    {
+      test: allCssRegex,
+      // Don't consider CSS imports dead code even if the
+      // containing package claims to have no side effects.
+      // Remove this when webpack adds a warning or an error for this.
+      // See https://github.com/webpack/webpack/issues/6571
+      sideEffects: true,
+      use: [
+        {
+          loader: styleInjector === 'mini-css-extract-plugin' ? MiniCssExtractPlugin.loader : styleLoaderPath,
+          options: injectorOptions,
+        },
+        {
+          loader: cssLoader,
+          options: {
+            importLoaders: 3,
+            modules: {
+              auto: modules,
+              getLocalIdent: getCSSModuleLocalIdent, // pretty class names
+            },
+
+            ...cssLoaderOptions,
+          },
+        },
+        postcssOptions && {
+          loader: postcssLoader,
+          options: {
+            postcssOptions,
+          },
+        },
+      ],
+      rules: [
+        // dialects
+        {
+          test: sassRegex,
+          use: [
+            {
+              loader: resolveUrlLoader,
+              options: resolveUrlOptions,
+            },
+            {
+              loader: sassLoader,
+              options: sassLoaderOptions,
+            },
+          ],
+        },
+        {
+          test: lessRegex,
+          use: [
+            {
+              loader: resolveUrlLoader,
+              options: resolveUrlOptions,
+            },
+            {
+              loader: lessLoader,
+              options: lessLoaderOptions,
+            },
+          ],
+        },
+      ],
+    },
+  ];
+
+  const stylePlugins: WebpackPluginInstance[] = [
+    // setting as 'any' because of this error:
+    // ts2321 - Excessive stack depth comparing types 'WebpackPluginInstance' and 'MiniCssExtractPlugin'.
+    styleInjector === 'mini-css-extract-plugin' && (new MiniCssExtractPlugin(miniCssOptions) as any),
+  ].filter(Boolean);
+
+  return {
+    styleLoaders,
+    stylePlugins,
+  };
+};
+
+/** loaders for pure css, including css-modules support and postcss but without and dialects (sass, less) */
+export const makeCssLoaders = ({
```

Comment:
let's rename it to generate instead of make?

Reply:
done